ovs/datapath.go

```diff
@@ -0,0 +1,338 @@
+	// Start by building the args
+	if dpName == "" {
+		return nil, errMissingDataPathName
+	}
+
+	args := []string{"ct-get-limits", dpName}
+
+	zoneParam := getZoneString(zones)
+	if zoneParam != "" {
+		args = append(args, zoneParam)
+	}
+
+	// call the cli
+	results, err := dp.CLI.Exec(args...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Process the results
+	entries := strings.Split(string(results), "\n")
+	ctOut := &ConnTrackOutput{}
+
+	r, err := regexp.Compile(`default`)
+	if err != nil {
+		return nil, err
+	}
+
+	// First start extracting the default conntrack limit setup
+	// If found the default value is removed from the entries
+	for i, entry := range entries {
+		if r.MatchString(entry) {
+			ctOut.defaultLimit = make(Zone)
+			limit, err := strconv.Atoi(strings.Split(entry, "=")[1])
+			if err != nil {
+				return nil, err
+			}
+			ctOut.defaultLimit["default"] = uint64(limit)
+			// As the default has been found let's remove it
+			entries = append(entries[:i], entries[i+1:]...)
+		}
+	}
+
+	// Now process the zones setup
+	for _, entry := range entries {
+		fields := strings.Split(entry, ",")
+		z := make(Zone)
+		for _, field := range fields {
+			buf := strings.Split(field, "=")
+			val, _ := strconv.Atoi(buf[1])
+			z[buf[0]] = uint64(val)
+		}
+		ctOut.zones = append(ctOut.zones, z)
+	}
+
+	return ctOut, nil
+}
+
+// SetCTLimits set the limit for a specific zone or globally.
+// Only one zone or default can be set up at once as the cli allows.
+// Examples of commands it wrapps:
+// sudo ovs-dpctl ct-set-limits system@ovs-system zone=331,limit=1000000
+// sudo ovs-dpctl ct-set-limits system@ovs-system default=1000000
+func (dp *DataPathService) SetCTLimits(dpName string, zone map[string]uint64) (string, error) {
+	// Sanitize the input
+	if dpName == "" {
+		return "", errMissingDataPathName
+	}
+	argsStr, err := ctSetLimitsArgsToString(zone)
+	if err != nil {
+		return "", err
+	}
+	// call the cli
+	argsCLI := []string{"ct-set-limits", dpName, argsStr}
+	results, err := dp.CLI.Exec(argsCLI...)
+
+	return string(results), err
+}
+
+// DelCTLimits deletes limits setup for zones. It takes the Datapath name
+// and zones to delete the limits.
+// sudo ovs-dpctl  ct-del-limits system@ovs-system  zone=40,4
+func (dp *DataPathService) DelCTLimits(dpName string, zones []uint64) (string, error) {
+	if dpName == "" {
+		return "", errMissingDataPathName
+	}
+	if len(zones) < 1 {
+		return "", errMissingMandatoryZone
+	}
+
+	var firstZone uint64
+	firstZone, zones = zones[0], zones[1:]
+	zonesStr := "zone=" + strconv.FormatUint(firstZone, 10)
+	for _, z := range zones {
+		zonesStr += "," + strconv.FormatUint(z, 10)
+	}
+
+	// call the cli
+	argsCLI := []string{"ct-del-limits", dpName, zonesStr}
+	results, err := dp.CLI.Exec(argsCLI...)
+
+	return string(results), err
+}
+
+// ctSetLimitsArgsToString is function to help formating and sanatizing an input
```

Comment:
nit: "ctSetLimitsArgsToString is function to help" -> "ctSetLimitsArgsToString helps[...]